pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -170,17 +170,25 @@ func (s *Server) initK8SConfigStore(args *PilotArgs) error {
 		})
 		if features.EnableGatewayAPIDeploymentController {
 			s.addTerminatingStartFunc("gateway deployment controller", func(stop <-chan struct{}) error {
-				// We can only run this if the Gateway CRD is created
-				if configController.WaitForCRD(gvk.KubernetesGateway, stop) {
-					tagWatcher := revisions.NewTagWatcher(s.kubeClient, args.Revision)
-					controller := gateway.NewDeploymentController(s.kubeClient, s.clusterID,
-						s.webhookInfo.getWebhookConfig, s.webhookInfo.addHandler, tagWatcher, args.Revision)
-					// Start informers again. This fixes the case where informers for namespace do not start,
-					// as we create them only after the CRD is created.
-					s.kubeClient.RunAndWait(stop)
-					go tagWatcher.Run(stop)
-					controller.Run(stop)
-				}
+				leaderelection.
```

Comment:
Do we want to do a leader election for each controller - or a leader election per revision that can be used by all controllers that need one ? Is this the only per-revision election ?